tools/restatectl/src/commands/partition/list.rs

```diff
@@ -79,27 +107,85 @@ pub async fn list_partitions(
                     let host_node =
                         GenerationalNodeId::new(host.id, host.generation.expect("generation"));
                     let details = PartitionListEntry { host_node, status };
-                    partitions.entry(partition_id).or_default().push(details);
+                    partitions.push((partition_id, details));
                 }
             }
         }
     }
+
     // Show information organized by partition and node
     let mut partitions_table = Table::new_styled();
     partitions_table.set_styled_header(vec![
         "P-ID",
         "NODE",
         "MODE",
         "STATUS",
-        "APPLIED",
-        "PERSISTED",
         "LEADER",
         "EPOCH",
+        "SEQUENCER",
+        "APPLIED",
+        "PERSISTED",
         "SKIPPED",
         "LAST-UPDATE",
     ]);
-    for (partition_id, processors) in partitions {
-        for processor in processors {
+
+    partitions
+        .into_iter()
+        .sorted_by(|a, b| match opts.sort {
+            SortMode::Partition => a.0.cmp(&b.0),
+            SortMode::Node => {
+                a.1.host_node
+                    .cmp(&b.1.host_node)
+                    .then_with(|| a.0.cmp(&b.0))
+            }
+            SortMode::Active => {
+                a.1.status
+                    .effective_mode
+                    .cmp(&b.1.status.effective_mode)
+                    .then_with(|| a.1.host_node.cmp(&b.1.host_node))
```

Comment:
Is there are reason why `Active` uses `Node` as secondary sort criterium compared to `Node`?

Reply:
Yes, this was intentional - when testing locally, it made more sense to me to order the active PPs by their node, rather than by partition id. But this is the _partitions_ list so you're probably more focused on finding a partition rather. I'll take this out. We'll have a separate node-focused view that also displays per-node functions for the use I had in mind.